crates/wasmparser/benches/benchmark.rs

```diff
@@ -263,14 +268,625 @@ fn validate_benchmark(c: &mut Criterion) {
     let mut inputs = collect_benchmark_inputs();
     // Filter out all benchmark inputs that fail to validate via `wasmparser`.
     inputs.retain(|input| validator().validate_all(&input.wasm).is_ok());
-    c.bench_function("validate benchmark", move |b| {
+    c.bench_function("validate all test modules", move |b| {
         b.iter(|| {
             for input in &mut inputs {
                 validator().validate_all(&input.wasm).unwrap();
             }
         })
     });
+
+    for file in std::fs::read_dir("benches").unwrap() {
+        let file = file.unwrap();
+        let path = file.path();
+        if path.extension().and_then(|s| s.to_str()) != Some("wasm") {
+            continue;
+        }
+        let name = path.file_stem().unwrap().to_str().unwrap();
```

Comment:
You may want to pack these into a helper function and move the call into the `c.bench_function` call since otherwise these are always called independent of the benchmark filters. Putting all the setup code into the `bench_function` closure help with profiling single benchmarks, for example when using `flamegraph`.

Reply:
I don't think that the `PathBuf` manipulation here is worth factoring out, but I used `once_cell` to factor out some other bits and pieces into more lazy computations to ensure they happen at most once and only if necessary.

Reply:
That's a cool application of `once_cell`. Gotta do the same in `wasmi`. :)